core/Pos/Core/Genesis/Testnet.hs

```diff
@@ -0,0 +1,217 @@
+-- | Generation of genesis data for testnet.
+
+module Pos.Core.Genesis.Testnet
+       ( GeneratedGenesisData (..)
+       , generateSecrets
+
+       , generateFakeAvvm
+       , generateSecretsAndAddress
+
+       , generateFakeAvvmGenesis
+       , generateTestnetData
+
+       , generateTestnetOrMainnetData
+       ) where
+
+import           Universum
+
+import           Crypto.Random              (MonadRandom, getRandomBytes)
+import qualified Data.HashMap.Strict        as HM
+import qualified Data.Map.Strict            as Map
+import           Serokell.Util.Verify       (VerificationRes (..), formatAllErrors,
+                                             verifyGeneric)
+
+import           Pos.Binary.Class           (asBinary, serialize')
+import           Pos.Core.Address           (Address, IsBootstrapEraAddr (..),
+                                             addressHash, deriveLvl2KeyPair,
+                                             makePubKeyAddressBoot, makeRedeemAddress)
+import           Pos.Core.Coin              (coinPortionToDouble, mkCoin,
+                                             unsafeIntegerToCoin)
+import qualified Pos.Core.Genesis.Constants as Const
+import           Pos.Core.Genesis.Types     (AddrDistribution, BalanceDistribution (..),
+                                             FakeAvvmOptions (..),
+                                             GenesisInitializer (..),
+                                             GenesisWStakeholders (..),
+                                             TestnetBalanceOptions (..),
+                                             TestnetDistribution (..))
+import           Pos.Core.Vss               (VssCertificate, VssCertificatesMap,
+                                             mkVssCertificate)
+import           Pos.Crypto                 (EncryptedSecretKey, RedeemPublicKey,
+                                             SecretKey, VssKeyPair, deterministic,
+                                             emptyPassphrase, keyGen, randomNumberInRange,
+                                             redeemDeterministicKeyGen, safeKeyGen,
+                                             toPublic, toVssPublicKey, vssKeyGen)
+
+-- | Data generated by @genTestnetOrMainnetData@ using genesis-spec.
+data GeneratedGenesisData = GeneratedGenesisData
+    { ggdNonAvvmDistr     :: ![AddrDistribution]
+    -- ^ Address distribution for non avvm addresses
+    , ggdBootStakeholders :: !GenesisWStakeholders
+    -- ^ Set of boot stakeholders (richmen addresses or custom addresses)
+    , ggdGtData           :: !VssCertificatesMap
+    -- ^ Genesis vss data (vss certs of richmen)
+    , ggdSecretKeys       :: !(Maybe [(SecretKey, EncryptedSecretKey, VssKeyPair)])
+    -- ^ Secret keys for non avvm addresses
+    , ggdFakeAvvmSeeds    :: !(Maybe [ByteString])
+    -- ^ Fake avvm seeds (needed only for testnet)
+    }
```

Comment:
There's non avvm distribution, fake avvm seeds. What about real avvm stuff?

Reply:
@avieth we don't generate real avvm stuff, it's known, but we don't know real avvm seeds, only redeem public keys

Reply:
Ok. So this `GeneratedGenesisData` is useful / relevant only for testnet after all, right? We would never use this to construct a "real" genesis with "real" avvm data?